### ISSUE TITLE
Support for continuous integration (CI)

### ISSUE THREAD

**Issue by Sanjo** [xolvio/meteor-jasmine#335]
Abernix:
> I guess dispatch:mocha-phantomjs is using dispatch:phantomjs-tests to accomplish this?

Yeah, dispatch:phantomjs-tests is similar to what we had with Velocity. The only difference is that it is executed from the runTests function.

I've just submitted a [proposal for the CI support in the Meteor testing API](https://github.com/meteor/meteor/issues/6809).

I think how it is implemented in dispatch:mocha-phantomjs makes sense in the current state. So we can basically copy that.

But I think it's not necessary to have a separate package just for phantomjs. I would rather have the options:

* CI = true/false
* BROWSER = null/phantomjs/chrome/firefox (The browser the tests should run in. Initially just null or phantomjs is supported. null means that you need to open the browser yourself)

Those options can be passed via environment variables.

**Comment by rjsmith** [comment]
Hi,

Look forward to this. It would be great if there was a server integration test - only CI option / configuration setup. So I can run a bunch of server integration tests on CI without needing to bring in a dependency on phantomJS.

**Comment by hg-stolle** [comment]
+1
I'd like to see this feature implemented.

**Comment by hg-stolle** [comment]
Any progress with this issue?

My company is developing a Meteor application with Jasmine as test framework. However, we cannot use the app in production, if there is no support for CI. Another option would be to switch to another test framework, but that would mean we have to rewrite the tests for that framework, and I would like to avoid that.

**Comment by Sanjo** [comment]
Hi, as stated in the README

> This package is no longer maintained!

> See [Migrate to Mocha](https://github.com/xolvio/meteor-jasmine/wiki/Migrate-to-Mocha) for how to use the Jasmine expect and spyOn APIs with Mocha.

**Comment by hg-stolle** [comment]
Thanks for the hint. I haven't seen that yet, because the deprecation message was added after I stumbled upon this issue.

Sadly, this basically means that Jasmine is not and will never be an option for CI of a Meteor application. :(

**Comment by Sanjo** [comment]
> Sadly, this basically means that Jasmine is not and will never be an option for CI of a Meteor application. :(

Not necessarily. It just means that I won't do it. If you really need it, you can do the same as https://github.com/DispatchMe/meteor-mocha-phantomjs for Jasmine. But migrating to Mocha is really not a big deal and in my view you get some subtle benefits. If you follow my migration guide, you will likely not need to change much in your tests.

**Comment by rjsmith** [comment]
Hi,
Just wanted to pop in here and say that I was in exactly the same situation, migrating a Meteor 1.2 project with 200+ Velocity jasmine server integration tests, to Meteor 1.3.  With Jonas's help and the info on the Migration page above, I was able to get all of my tests successfully running using the new Meteor test framework. The same test suite now runs 25% faster too on my CI instance on codeship:

`meteor test --once --full-app --driver-package dispatch:mocha-phantomjs --settings my-settings.json"`

I wrote up a summary of the issues I encountered and how I solved them here:

https://forums.meteor.com/t/solved-v1-3-velocity-integration-tests-with-sanjo-jasmine-no-longer-working-after-upgrade-from-1-2-1-to-1-3/20477/9?u=rsbatech

@hg-stolle : Hope that helps you....

**Comment by rjsmith** [comment]
To clarify: that Meteor forums post got me running on sanjo:jasmine on Meteor 1.3 . But when it got deprecated, I switched to the mocha test driver instead using the migration guide.  So I did the upgrade from Velocity + Jasmine on Meteor 1.2 to Mocha-PhantomJS + Meteor 1.3 in two steps.

**Comment by hg-stolle** [comment]
@rjsmith: Thanks anyway, but I've already migrated my tests to Mocha before I saw your summary.
